resin/models/device.py

```diff
@@ -715,7 +715,10 @@ def move(self, uuid, app_name):
         device = self.get(uuid)
         application = self.application.get(app_name)
 
-        if device['device_type'] != application['device_type']:
+        device_dev_type = filter(lambda dev_type: dev_type['slug'] == device['device_type'], self.config.get_device_types())[0]
```

Comment:
NIT: how about calling `self.config.get_device_types()` once (instead of twice) and keeping the result in a variable?